2024-06-17/main.py

```diff
@@ -0,0 +1,201 @@
+    return getUser(id)
+
+
+# Endpoint for getting ALL USERS
+@router.get("/get/all")
+def get_all_users(limit=10) -> list[dict]:
+    return getAllUsers(limit)
+
+
+# Endpoint for GETTING an EXISTING USER
+@router.get("/get/user")
+def get_user(id) -> dict:
+    return getUser(id)
+
+
+# Endpoint for UPDATING an EXISTING USER
+@router.patch("/update/user")
+def update_user(id, name=None, address=None, email=None, phone=None):
+    user = getUser(id)
+    if user:
+        return updateUser(
+            id,
+            user[id]["user_name"] if name is None else name,
+            user[id]["user_address"] if address is None else address,
+            user[id]["user_email"] if email is None else email,
+            user[id]["user_phone"] if phone is None else phone,
+        )
+    else:
+        raise HTTPException(status_code=404, detail="No such user.")
+
+
+# Endpoint for DELETING a USER
+@router.delete("/delete/user")
+def delete_user(id) -> str:
+    return deleteUser(id)
+
+
+# Method for CREATING a NEW USER
+def insertUser(name, address, email, phone) -> str:
+    user_id = uuid.uuid4().hex
+    params = (user_id, name, address, email, phone)
+    cur.execute(
+        "INSERT INTO users(id, name, address, email, phone) VALUES(?, ?, ?, ?, ?)",
+        params,
+    )
+    return user_id
+
+
+# Method for GETTING ALL USERS
+def getAllUsers(limit=10) -> list[dict]:
+    res = cur.execute("SELECT * FROM users LIMIT ?", (limit,))
+    users = res.fetchall()
+    if users is not None:
+        all_users = []
+        for user in users:
+            all_users.append(
+                {
+                    "user_id": user[0],
+                    "user_name": user[1],
+                    "user_address": user[2],
+                    "user_email": user[3],
+                    "user_phone": user[4],
+                }
+            )
+        return all_users
+    else:
+        return []
+
+
+# Method for GETTING an EXISTING USER
+def getUser(id) -> dict:
+    res = cur.execute("SELECT * from users WHERE id = ?", (id,))
+    user = res.fetchone()
+    if user is not None:
+        return_user = {
+            user[0]: {
+                "user_id": user[0],
+                "user_name": user[1],
+                "user_address": user[2],
+                "user_email": user[3],
+                "user_phone": user[4],
+            }
+        }
+        return return_user
+    else:
+        return None
+
+
+# Method for UPDATING an EXISTING USER
+def updateUser(id, user_name, user_add, user_email, user_phone) -> dict:
+
+    res = cur.execute(
+        "UPDATE users SET name = ?, email = ?, address = ?, phone = ? WHERE id = ?",
+        (user_name, user_email, user_add, user_phone, id),
+    )
+    db.commit()
+
+    return getUser(id)
+
+
+# Method for DELETING a USER
+def deleteUser(id) -> str:
+    res = cur.execute("DELETE FROM users WHERE id = ?", (id,))
+    db.commit()
+    return f"User {id} deleted"
```

Comment:
If you attempt to delete a user whose ID doesn't exist, what happens?

Reply:
Implemented if statement to check if user exists first.